Dockerfile-alpine

```diff
@@ -0,0 +1,168 @@
+
+# Download the glibc for Alpine .apks, and install
+RUN wget \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BASE_URL/$ALPINE_GLIBC_PACKAGE_VERSION/$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+RUN apk add \
+    "$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+RUN /usr/glibc-compat/bin/localedef --force --inputfile POSIX --charmap UTF-8 "$LANG" || true
+RUN echo "export LANG=$LANG" > /etc/profile.d/locale.sh
+
+# Cleanup glibc
+RUN rm "/etc/apk/keys/sgerrand.rsa.pub"
+RUN rm "/root/.wget-hsts"
+RUN rm \
+    "$ALPINE_GLIBC_BASE_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_BIN_PACKAGE_FILENAME" \
+    "$ALPINE_GLIBC_I18N_PACKAGE_FILENAME"
+
+#-------------------------------------------------------------------------------
+
+# Build libvips
+RUN wget -O- https://github.com/libvips/libvips/releases/download/v${VIPS_VERSION}/vips-${VIPS_VERSION}.tar.gz | tar xzC /tmp
+RUN cd /tmp/vips-${VIPS_VERSION} && \
+    ./configure \
+        --prefix=${VIPS_DIR} \
+        --disable-dependency-tracking \
+        --enable-debug=no \
+        --enable-fast-install \
+        --enable-gtk-doc-html \
+        --enable-introspection=no \
+        --enable-pyvips8 \
+        --enable-silent-rules \
+        --enable-static \
+    && make \
+    && make install \
+    && make -s install-strip
+
+#-------------------------------------------------------------------------------
+
+# Build Imaginary
+RUN mkdir -p ${GOPATH}/src
+RUN wget -O- https://github.com/h2non/imaginary/archive/v${IMAGINARY_VERSION}.tar.gz | tar xzC ${GOPATH}/src
+RUN cd ${GOPATH}/src/imaginary-${IMAGINARY_VERSION} && \
+    go get -u golang.org/x/net/context && \
+    go get -u github.com/golang/dep/cmd/dep && \
+    go get -u github.com/rs/cors && \
+    go get -u gopkg.in/h2non/filetype.v1 && \
+    go get -u github.com/throttled/throttled && \
+    dep ensure && \
+    go build -o $GOPATH/bin/imaginary
+
+#-------------------------------------------------------------------------------
+
+FROM alpine:3.9
+
+ENV PERSISTENT_DEPS \
+    ca-certificates \
+    cairo \
+    expat \
+    fftw \
+    fontconfig \
+    giflib \
+    glib \
+    gobject-introspection \
+    imagemagick \
+    lcms2 \
+    libexif \
+    libgsf \
+    libintl \
+    libjpeg-turbo \
+    libpng \
+    librsvg \
+    libwebp \
+    openexr \
+    openssl \
+    orc \
+    pango \
+    poppler-glib \
+    tiff
+
+RUN apk upgrade --update
+RUN apk add --virtual .persistent-deps $PERSISTENT_DEPS
+RUN echo "@testing http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
+RUN apk add libimagequant@testing
+RUN rm -rf /var/cache/apk/*
+
+COPY --from=build /vips/lib/ /usr/local/lib
+COPY --from=build /go/bin/imaginary /usr/bin/imaginary
+
+ENV PORT 9000
+EXPOSE 9000
```

Comment:
Perhaps `EXPOSE $PORT`

Reply:
> > @Dynom said: However the high-confidence part is then missing, without a proper functional test-covered result.
> 
> A small e2e suite based on starting a server and running some curl/etc could be a start?
> 
> > @Dynom said: I'm open to ideas on how to solve that missing piece :-/
> 
> I thought about this as well. Can't help but feel that these could be parallel efforts until later.

I fear as much. Unfortunately that means a ton of work in the functional-testing area. Something I would love to have, but it's not the most glorious work.

And thanks for the review effort @jbergstroem 💪🏾